src/main/java/teammates/ui/webapi/action/PostStudentProfilePictureAction.java

```diff
@@ -0,0 +1,74 @@
+package teammates.ui.webapi.action;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.Part;
+
+import org.apache.http.HttpStatus;
+
+import teammates.common.datatransfer.attributes.StudentProfileAttributes;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Const;
+import teammates.common.util.GoogleCloudStorageHelper;
+import teammates.ui.webapi.output.StudentProfilePictureResults;
+
+/**
+ * Action: saves the file information of the profile picture that was just uploaded.
+ */
+public class PostStudentProfilePictureAction extends Action {
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        if (!userInfo.isStudent) {
+            throw new UnauthorizedAccessException("Student privilege is required to access this resource.");
+        }
+    }
+
+    @Override
+    public ActionResult execute() {
+        try {
+            Part image = extractProfilePicture();
+            byte[] imageData = new byte[(int) image.getSize()];
+            if (imageData == null) {
```

Comment:
This does not help. If `image` itself is already null, the NPE is triggered before this line.

Reply:
`imageData` can't be null as discussed, so will be removing the check.